src/scripts/generate-heading-ids.ts

```diff
@@ -135,27 +161,31 @@ const traverseHeaders = (path, doc = "", write = false) => {
 
     console.log(`>>> processing ${file}`)
     curLevel = [0, 0, 0]
-    const content = fs.readFileSync(file, "utf8")
-    const lines = content.split("\n")
+    const content: string = fs.readFileSync(file, "utf8")
+    const lines: Array<string> = content.split("\n")
     const updatedLines = addHeaderIDs(lines, write)
     if (write) {
       fs.writeFileSync(file, updatedLines.join("\n"))
     }
   })
   if (!write) {
-    console.log(toc)
+    console.log(headersToChange)
   }
 }
 
-const addHeaderIDsForDir = (path) => {
+const addHeaderIDsForDir = (path: string) => {
   if (path.includes("translations")) {
     throw new Error(`Heading ID generation is intended for English files only.`)
   }
   const fullPath = `src/content/${path}`
-  traverseHeaders(fullPath, null, false)
-  traverseHeaders(fullPath, null, true)
+  traverseHeaders(fullPath, { write: false })
```

Comment:
**question**: should there be functionality to modify all the markdown files in a directory?
Currently, `headersToChange` just keeps track of header-number in a single markdown file.

Reply:
Hmm good question, that could be interesting. @minimalsm what do you think? (as I don't use very often this script). Could be handy

---
Note: either way, let's not include that potential feature in this PR.

Reply:
I never really use this either.